pkg/controller/certificates/clustertrustbundlepublisher/publisher_test.go

```diff
@@ -0,0 +1,337 @@
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testBundleName,
+					},
+					Spec: certificatesv1alpha1.ClusterTrustBundleSpec{
+						SignerName:  testSignerName,
+						TrustBundle: string(testCAProvider.CurrentCABundleContent()),
+					},
+				},
+				&certificatesv1alpha1.ClusterTrustBundle{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test.test/testSigner:name2",
+					},
+					Spec: certificatesv1alpha1.ClusterTrustBundleSpec{
+						SignerName:  testSignerName,
+						TrustBundle: string(testCAProvider.CurrentCABundleContent()),
+					},
+				},
+			},
+			expectActions: checkDeletedTestSignerBundle("test.test/testSigner:name2"),
+		},
+		{
+			name: "CTB at the correct state - noop",
+			existingCTBs: []runtime.Object{
+				&certificatesv1alpha1.ClusterTrustBundle{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "nosigner",
+					},
+					Spec: certificatesv1alpha1.ClusterTrustBundleSpec{
+						TrustBundle: "somedatahere",
+					},
+				},
+				&certificatesv1alpha1.ClusterTrustBundle{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "signer:one",
+					},
+					Spec: certificatesv1alpha1.ClusterTrustBundleSpec{
+						SignerName:  "signer",
+						TrustBundle: "signerdata",
+					},
+				},
+				&certificatesv1alpha1.ClusterTrustBundle{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: testBundleName,
+					},
+					Spec: certificatesv1alpha1.ClusterTrustBundleSpec{
+						SignerName:  testSignerName,
+						TrustBundle: string(testCAProvider.CurrentCABundleContent()),
+					},
+				},
+			},
+			expectActions: func(t *testing.T, actions []clienttesting.Action) {
+				actions = filterOutListWatch(actions)
+				if len(actions) != 0 {
+					t.Fatalf("expected no actions, got %v", actions)
+				}
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			testCtx := ktesting.Init(t)
+
+			fakeClient := fakeKubeClientSetWithCTBList(t, testSignerName, tt.existingCTBs...)
+
+			ctbInformer := setupSignerNameFilteredCTBInformer(fakeClient, testSignerName)
+			go ctbInformer.Run(testCtx.Done())
+
+			syncWait := make(chan struct{})
+			go func() {
+				for !ctbInformer.HasSynced() {
+					select {
+					case <-testCtx.Done():
+						return
+					case <-time.After(10 * time.Millisecond):
+					}
+				}
+				syncWait <- struct{}{}
+			}()
+
+			select {
+			case <-testCtx.Done():
+				t.Fatalf("timed out waiting for informer to sync")
+			case <-syncWait:
+			}
+
+			p := &ClusterTrustBundlePublisher{
+				client:          fakeClient,
+				ctbLister:       certalpha1listers.NewClusterTrustBundleLister(ctbInformer.GetIndexer()),
+				ctbListerSynced: func() bool { return true },
+				signerName:      testSignerName,
+				ca:              testCAProvider,
+
+				queue: workqueue.NewTypedRateLimitingQueueWithConfig(
+					workqueue.DefaultTypedControllerRateLimiter[string](),
+					workqueue.TypedRateLimitingQueueConfig[string]{
+						Name: "test_root_ca_cert_publisher_cluster_trust_bundles",
+					},
+				),
+			}
```

Comment:
Can we re-use `NewClusterTrustBundlePublisher`?

Reply:
turns out we can :)